src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java

```diff
@@ -173,12 +191,32 @@ private long getSubmitBlockedTime() {
         return submitService.getSubmitTimer().durationInSeconds();
     }
 
-    private void sendTimingWarningToSentry(SimpleTimer timer) {
+    private Map<String, Long> getSlowRequestThresholds() {
+        Map<String, Long> slowRequestThresholds = new HashMap<>();
+        slowRequestThresholds.put("answer", Long.valueOf(5 * 1000));
+        slowRequestThresholds.put("submit-all", Long.valueOf(20 * 1000));
+        slowRequestThresholds.put("navigate_menu", Long.valueOf(20 * 1000));
+        return slowRequestThresholds;
+    }
+
+    private String getSentryMessageForSlowRequest(String category) {
+        Map<String, String> messages = new HashMap<>();
+        messages.put(EXTREMELY_SLOW_REQUEST, "This request took a long time");
```

Comment:
These two messages are basically impossible to tell apart if you dont already know which is which. Is it possible to add more meaningful messages (that indicate which is slower for example)

Reply:
To not disrupt current workflows, the goal is to keep the message "This request took a long time". With that in mind, only the new message can be improved. I updated but am open to suggestions.